### PR TITLE
WIP: add pointer option

### PR DESCRIPTION
**Pull request by ElieDeloumeau**
related to #103 

(squash will be done when validation complete)

### CODE REVIEW

File: questionary/prompts/common.py [hunk 1]
```diff
@@ -188,6 +188,7 @@ def __init__(
         self,
         choices: Sequence[Union[str, Choice, Dict[str, Any]]],
         default: Optional[Union[str, Choice, Dict[str, Any]]] = None,
+        pointer: str = DEFAULT_SELECTED_POINTER,
```
Comment (kiancross) [inline questionary/prompts/common.py:191]: Can you add a description in the docstring for this argument?

I think perhaps we could remove the `use_pointer` option and just set `pointer` to `None` to indicate that the pointer should be hidden.

**Review by kiancross — CHANGES_REQUESTED** [review]
Thanks for the PR. It would also be nice to add this option to `checkbox`.

File: questionary/prompts/common.py (hunk 1 above)
Comment (ElieDeloumeau) [reply questionary/prompts/common.py:191]: Sure it would be less redundant

**Comment by ElieDeloumeau** [comment]
> Thanks for the PR. It would also be nice to add this option to `checkbox`.

Done 👍

File: questionary/prompts/common.py [hunk 2]
```diff
@@ -180,27 +180,27 @@ class InquirerControl(FormattedTextControl):
     use_indicator: bool
     use_shortcuts: bool
     use_arrow_keys: bool
-    use_pointer: bool
+    pointer: str
```
Comment (kiancross) [inline questionary/prompts/common.py:183]:
```suggestion
    pointer: Optional[str]
```

File: questionary/prompts/common.py [hunk 3]
```diff
@@ -180,27 +180,27 @@ class InquirerControl(FormattedTextControl):
...
     pointed_at: int
     is_answered: bool
 
     def __init__(
         self,
         choices: Sequence[Union[str, Choice, Dict[str, Any]]],
         default: Optional[Union[str, Choice, Dict[str, Any]]] = None,
+        pointer: str = DEFAULT_SELECTED_POINTER,
```
Comment (kiancross) [inline questionary/prompts/common.py:191]:
```suggestion
        pointer: Optional[str] = DEFAULT_SELECTED_POINTER,
```

File: questionary/prompts/rawselect.py [hunk 4]
```diff
@@ -13,6 +13,7 @@ def rawselect(
     choices: Sequence[Union[str, Choice, Dict[str, Any]]],
     default: Optional[str] = None,
     qmark: str = DEFAULT_QUESTION_PREFIX,
+    pointer: str = DEFAULT_SELECTED_POINTER,
```
Comment (kiancross) [inline questionary/prompts/rawselect.py:16]:
```suggestion
    pointer: Optional[str] = DEFAULT_SELECTED_POINTER,
```

File: questionary/prompts/select.py [hunk 5]
```diff
@@ -19,11 +23,11 @@ def select(
     choices: Sequence[Union[str, Choice, Dict[str, Any]]],
     default: Optional[Union[str, Choice, Dict[str, Any]]] = None,
     qmark: str = DEFAULT_QUESTION_PREFIX,
+    pointer: str = DEFAULT_SELECTED_POINTER,
```
Comment (kiancross) [inline questionary/prompts/select.py:26]:
```suggestion
    pointer: Optional[str] = DEFAULT_SELECTED_POINTER,
```

**Review by kiancross — CHANGES_REQUESTED** [review]
I think the `pointer` argument needs to be made optional so that `None` can be passed.

There will also be a problem if `pointer` is longer than a single character. I have tested a quick fix for this and I will push it to this PR if you are ok with that.

**Comment by ElieDeloumeau** [comment]
> I think the `pointer` argument needs to be made optional so that `None` can be passed.
> 
> There will also be a problem if `pointer` is longer than a single character. I have tested a quick fix for this and I will push it to this PR if you are ok with that.

Of course you can 😃

**Comment by ElieDeloumeau** [comment]
> There will also be a problem if `pointer` is longer than a single character. I have tested a quick fix for this and I will push it to this PR if you are ok with that.

Is this really a problem? The list is shifted but it doesn't matter?!

**Comment by kiancross** [comment]
> Is this really a problem? The list is shifted but it doesn't matter?!

I think it is better if the choices remain aligned. Let me know if you are happy with the changes and then I will merge.

**Review by kiancross — APPROVED** [review]

**Pull request merged**